Sources/StreamChat/Models/Attachment.swift

```diff
@@ -4,82 +4,55 @@
 
 import Foundation
 
-/// A type representing a chat message attachment. `ChatMessageAttachment` is an immutable snapshot of a
-/// chat message attachment entity at the given time.
-///
-/// - Note: `ChatMessageAttachment` is a typealias of `_ChatMessageAttachment` with default extra data.
-/// If you're using custom extra data, create your own typealias of `_ChatMessageAttachment`.
-///
-/// Learn more about using custom extra data in our [cheat sheet](https://github.com/GetStream/stream-chat-swift/wiki/Cheat-Sheet#working-with-extra-data).
-///
-public typealias ChatMessageAttachment = _ChatMessageAttachment<NoExtraData>
+/// A protocol for attachment model objects that are created from database `AttachmentDTO` objects and exposed.
+public protocol DatabaseAttachment {
+    var type: AttachmentType { get }
+    var id: AttachmentId? { get }
+}
```

Comment:
> OK, what If we called this ChatMessageAttachment since it's our SDK's output. What if the existing ChatMessageAttachment becomes AttachmentEnvelope?

Yep, makes sense.